test/loader-common.js

```diff
@@ -1275,6 +1337,38 @@ export const LoaderCommonFactory = ({
       }
     );
 
+    QUnit.test('chooses the previous part if not buffered and current is not independent', function(assert) {
+      loader.buffered_ = () => videojs.createTimeRanges();
+      const playlist = playlistWithDuration(50, {llhls: true});
+
+      loader.hasPlayed_ = () => true;
+      loader.currentTime_ = () => 46;
+      loader.syncPoint_ = null;
+
+      loader.playlist(playlist);
+      loader.load();
+      // in a real scenario this probably won't happen
+      // but it makes which segment/part is chosen easier.
+      loader.mediaIndex = 4;
+      loader.partIndex = 1;
+
+      playlist.segments[4].parts[1].independent = true;
+      const segmentInfo = loader.chooseNextRequest_();
+
+      assert.equal(segmentInfo.partIndex, 1, 'previous part');
```

Comment:
Doesn't this match the partIndex above?

Reply:
it does, but we also don't have anything buffered so it tries to choose sepmentIndex 4, partIndex 2 but that isn't independent so we choose the current partIndex. Let me see if I can find something better than this though.